agent/api/task/service_connect.go

```diff
@@ -65,3 +80,55 @@ type DNSConfigEntry struct {
 	HostName string `json:"hostName"`
 	Address  string `json:"address"`
 }
+
+// ExtractSCConfigFromACSAttachment finds the service connect attachment, and extracts the service connect container name and service connect config value.
+func ExtractSCConfigFromACSAttachment(acsTask *ecsacs.Task) (*ServiceConnectConfig, error) {
+	if acsTask.Attachments == nil {
+		return nil, nil
+	}
+	var scAttachment *ecsacs.Attachment
+	containerName := ""
+	for _, attachment := range acsTask.Attachments {
+		if aws.StringValue(attachment.AttachmentType) == serviceConnectAttType {
+			scAttachment = attachment
+			break
+		}
+	}
+	// no service connect attachment is found
+	if scAttachment == nil {
+		return nil, nil
+	}
+	// translate ecsacs.Attachment.Value to ServiceConnectConfig
+	scConfigValue := &ServiceConnectConfig{}
+	for _, property := range scAttachment.AttachmentProperties {
+		switch aws.StringValue(property.Name) {
+		case serviceConnectAttProSCConfigName:
+			s, _ := strconv.Unquote(string(*property.Value))
+			if err := json.Unmarshal([]byte(s), scConfigValue); err != nil {
+				return nil, err
+			}
+		case serviceConnectAttProSCContainerName:
+			containerName = aws.StringValue(property.Value)
+		}
+	}
+	scConfigValue.ContainerName = containerName
+	if err := ValidateSCConfig(scConfigValue); err != nil {
+		return nil, err
+	}
+	return scConfigValue, nil
+}
+
+func ValidateSCConfig(scConfig *ServiceConnectConfig) error {
+	if scConfig.ContainerName == "" {
+		return errors.Errorf("service connect config validation: no service connect ContainerName")
+	}
+
+	if len(scConfig.IngressConfig) == 0 {
```

Comment:
I wonder if we want to validate if each `IngressConfigEntry` contains the necessary fields. For example, if task has `awsvpc` network mode, then every `IngressConfigEntry` should have 
- either `interceptPort` + `listenerName` (for default experience)
- or `listenerPort` (for non-default experience)

Reply:
As mentioned offline, we can have services that are client only and they will not have any ingress config. This is valid.

Reply:
Please take a look at commit 8e69368bc5c1d7bd3ab71391103f4b6a69f06054 with updated test cases and validations. Thanks.